src/Task.php

```diff
@@ -0,0 +1,315 @@
+     * @param string $status
+     * @return boolean
+     */
+    private function canChangeStatus($status)
+    {
+        $statusChain = [
+            self::STATUS_NEW => [
+                self::STATUS_CANCELED, self::STATUS_INPROGRESS
+            ],
+            self::STATUS_INPROGRESS => [
+                self::STATUS_COMPLETE, self::STATUS_FAIL
+            ]
+        ];
+
+        if (!isset($statusChain[$this->status])) {
+            return false;
+        }
+
+        return in_array($status, $statusChain[$this->status]);
+    }
+
+    /**
+     * Проверяет, допускаетлся выполнение Действия
+     *
+     * @param string $action
+     * @return boolean
+     */
+    private function canRunAction($action)
+    {
+        $actionChain = [
+            self::STATUS_NEW => [
+                self::ACTION_CUSTOMER_APPROVE,
+                self::ACTION_CUSTOMER_CANCEL,
+                self::ACTION_PERFORMER_PENDING,
+                self::ACTION_PERFORMER_START
+            ],
+            self::STATUS_INPROGRESS => [
+                self::ACTION_CUSTOMER_CANCEL,
+                self::ACTION_CUSTOMER_COMPLETE,
+                self::ACTION_PERFORMER_REFUSE
+            ]
+        ];
+        if (!isset($actionChain[$this->status])) {
+            return false;
+        }
+
+        return in_array($action, $actionChain[$this->status]);
+    }
+
+    /**
+     * Возвращает список доступных `Действий` для текущего `Статуса`
+     * @return array Список доступных действий
+     */
+    private function avaiableActions()
+    {
+        $statusActions[self::STATUS_NEW] = [
+            self::ACTION_CUSTOMER_CANCEL,
+            self::ACTION_PERFORMER_PENDING,
+            self::ACTION_PERFORMER_START
+        ];
+
+        $statusActions[self::STATUS_INPROGRESS] = [
+            self::ACTION_CUSTOMER_COMPLETE,
+            self::ACTION_CUSTOMER_REFUSE,
+            self::ACTION_PERFORMER_DONE,
+        ];
+
+        return isset($statusActions[$this->status]) ? $statusActions[$this->status] : [];
+    }
+
+
+    /**
+     * Проверяет соответсвие запрашиваемого статуса к списку возможных статусов.
+     * 
+     * @property string $status Именованный статус задания.
+     * @return boolean true or false
+     */
+    private static function isStatusValid(string $status)
+    {
+        $statusList = array_values(self::StatusList());
+        if (in_array($status, $statusList)) {
+            return true;
+        } else {
+            echo "Состояние '{$status}' не входит с писок допустимых.\n";
+            echo "Не пытайтесь в ручную указывать состояние задачи, используйте доступные действия.\n";
+            return false;
+        }
+    }
+
+    /**
+     * Метод выполняет проверку, разрешено ли использовать запрашиваемый метод.
+     *
+     * @param string $action
+     * @return boolean
+     */
+    private static function isActionValid($action)
+    {
+        return in_array($action, self::ActionList()) ? true : false;
```

Comment:
Т.к. in_array итак возвращает bool, то можно просто сделать так:
```suggestion
        return in_array($action, self::ActionList());
```

Reply:
Убрал лишнее